### PR TITLE
Integrate keyboard shortcuts to quickly navigate to core pages

### PR DESCRIPTION
**Pull request by mariodian**
As requested in #91. Will probably need to use globalShortcuts in the future.

### CODE REVIEW

**Comment by morebrownies** [comment]
Hey @mariodian just thought of something else. It would be nice to have a shortcut to focus the address bar so that you don't have to use the mouse to access it. Possibly CMD + L?

File: js/main.js [hunk 1]
```diff
@@ -143,6 +143,59 @@ $('body').on('keypress', 'input', function(event) {
...
+//keyboard shortucts
+window.keyShortcuts = {
+	discover: 		'd',
+	myPage: 		'h',
+	customizePage: 	'e',
+	create: 		'n',
+	purchases: 		't',
+	sales: 			's',
+	cases: 			'b',
+	settings: 		','
```
Comment (jjeffryes) [inline js/main.js]: Why doesn't settings have a letter?

File: js/main.js [hunk 2]
```diff
@@ -143,6 +143,59 @@ $('body').on('keypress', 'input', function(event) {
   }
 });
 
+//keyboard shortucts
+window.keyShortcuts = {
+	discover: 		'd',
+	myPage: 		'h',
+	customizePage: 	'e',
```
Comment (jjeffryes) [inline js/main.js]: Become a Moderator should also have a shortcut.

**Comment by jjeffryes** [comment]
I noticed that if you open the nav menu to see the shortcuts, and then use a shortcut, it leaves the menu open. 

Can you add a line to the cleanup function in router.js that closes the nav menu if it's open?

File: js/main.js (hunk 1 above)
Comment (morebrownies) [inline js/main.js]: It has a comma

Comment (jjeffryes) [inline js/main.js]: Ah. Missed that. Can it be something easier to see? It's not very visible in the menu.

Comment (morebrownies) [inline js/main.js]: Could we go with ~ ?

Comment (mariodian) [inline js/main.js]: You can't write ~ without the shift key on mac. 

The reason I used comma is because mac users know it from Safari for "Preferences".

Comment (morebrownies) [inline js/main.js]: Oh right

Comment (jjeffryes) [inline js/main.js]: I'm a Mac user, I didn't know that. It must be the fact I only use my Mac sometimes, I've been tainted by dark forces. ;)

I think so far the majority of our users are on Windows, using patterns familiar to Mac users isn't going to be a benefit for most of them.

Any random letter is probably fine, maybe "e"?

Comment (mariodian) [inline js/main.js]: "E" is already used for "Customize page".

I'm not sure if windows users use any shortcuts to access Settings in their OS. In that case a mac-friendly shortcut wouldn't do any harm to them, but would be benefit for mac users. What key do linux users use?

**Comment by mariodian** [comment]
What if Purchases, Sales and Cases were cmd + 0, cmd + 1, cmd + 2 (they are on the same page anyway), so we could use "S" for Settings and "B" for Become Moderator?

Yes, I'll modify cleanup in router.

**Comment by morebrownies** [comment]
We should try to avoid setting hotkeys for temporary screens or one time experience screens such as Become a moderator or Become a store. Keyboard shortcuts are really intended to make getting to frequently used screens faster or for frequently used actions.

**Comment by mariodian** [comment]
Done.

Discover => cmd+d (d as discover)
My Page => cmd+h (as home)
Create Listing => cmd+n (as new)
Purchases => cmd+1 (first view of transaction page)
Sales => cmd+2 (second view of transaction page)
Cases => cmd+3 (third view of transaction page)
Settings => cmd+s
Address bar focus => cmd+l

**Comment by morebrownies** [comment]
Nice! I like it. The only one that concerns me is cmd + s since that is so widely used for saving. People may hit cmd + s on accident thinking it may save their listing or something and instead it will take them to settings.

**Comment by mariodian** [comment]
I was also thinking about cmd+g, like "Global settings", or cmd+p (Preferences), but that's usually used for printing. Perhaps something non-related such as cmd+z would work as well. Let me know what you think.

**Comment by morebrownies** [comment]
Let's go with cmd + G.
On Mar 11, 2016 12:08 PM, "Mario Dian" notifications@github.com wrote:

I was also thinking about cmd+g, like "Global settings", or cmd+p
(Preferences), but that's usually used for printing. Perhaps something
non-related such as cmd+z would work as well. Let me know what you think.

**Pull request merged**